[Bugfix:System] Remove user_id from Core

DuaaScripts opened the pull request:
### Please check if the PR fulfills these requirements:

* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant
* [ ] Screenshots are attached to Github PR if visual/UI changes were made

### What is the current behavior?
`Core.php` has unused `user_id` member variable

Partially addresses #11262

### What is the new behavior?
`user_id` has been removed

cjreed121 reviewed (changes requested):
This is not the desired change for this issue. Instead, you should only be removing the `$user_id` member variable from `Core.php` and then improving some of the type hints as they are not correct with their default values. No other PHP files should be need to be changed for this PR.

DuaaScripts commented:
@cjreed121 Thank you for the feedback!  I’ll remove the $user_id member variable from Core.php and fix the type hints as per your suggestion. I’ll update the PR shortly with these changes. @cjreed121

DuaaScripts commented:
@cjreed121 
I’ve made the requested changes:
- Removed the $user_id member variable from Core.php.
- Updated the type hints where they were incorrect or didn’t match the expected values.
- 
Please let me know if anything else needs to be improved, or if I missed anything! Looking forward to your feedback.

cjreed121 commented:
Please revert changes to all other files besides `Core.php` in this PR

DuaaScripts commented:
@cjreed121 
Done with the changes, ready for review

cjreed121 commented:
Please address the failing PHP lint test. Static analysis may also fail so make sure to run them locally first.

DuaaScripts commented:
@cjreed121 
I've tried to fix the PHP lint errors. waiting for your review.

cjreed121 commented:
Please fix PHP static analysis test and verify it passes locally. See https://submitty.org/developer/testing/linting_static_analysis#php-static-analysis for more details on how to run it and regenerate the baseline.

DuaaScripts commented:
@cjreed121 
I’ve fixed the PHPStan error and tested the code locally. Hopefully, the tests will pass now. 
Looking forward to your review! 😊

cjreed121 commented:
Please update the PR description to reflect new changes. Remove the "Fixes ..." since this does not address the whole issue since there will need to be a bit more done but we can get this merged first. Still mention the issue but with some other keyword than "Fixes" or "Closes".

DuaaScripts commented:
@cjreed121  Done!

cjreed121 commented:
The description changes have not been updated. The description says you removed a certain method when you did not. Please address this.

Also no need to ping me for every change and don't hit the update branch button unless you need to pull in other changes or resolve conflicts.
